vispy/scene/cameras/cameras.py

```diff
@@ -71,43 +197,191 @@ def interactive(self):
+        # Store center location
+        self._center_loc = [(b[0] + r / 2) for b, r in zip(bounds, ranges)]
+        # Set given params (since _set_range() might derive other values)
+        for name, value in self._given_params.items():
+            setattr(self, name, value)
+        # Let specific camera handle it
+        self._set_range()
+        # Overwrite given params
+        for name, value in self._given_params.items():
+            setattr(self, name, value)
+        # Finish reset
+        self._resetting = False
+        self.view_changed()
+    
+    def _set_range(self):
+        pass 
+    
+    def get_state(self):
+        """ Get the current view state of the camera
+        
+        Returns a dict of key-value pairs. The exact keys depend on the
+        camera. Can be passed to set_state() (of this or another camera
+        of the same type) to reproduce the state.
+        """
+        D = {}
+        for key in self._state_props:
+            D[key] = getattr(self, key)
+        return D
+    
+    def set_state(self, state=None, **kwargs):
+        """ Set the view state of the camera
+        
+        Should be a dict (or kwargs) as returned by get_state. It can
+        be an incomlete dict, in which case only the specified
+        properties are set.
         """
-        self._update_transform()
+        D = state or {}
+        D.update(kwargs)
+        for key, val in D.items():
+            if key not in self._state_props:
+                raise KeyError('Not a valid camera state property %r' % key)
+            setattr(self, key, val)
     
-    def connect(self):
-        self._viewbox.events.mouse_press.connect(self.view_mouse_event)
-        self._viewbox.events.mouse_release.connect(self.view_mouse_event)
-        self._viewbox.events.mouse_move.connect(self.view_mouse_event)
-        self._viewbox.events.mouse_wheel.connect(self.view_mouse_event)
-        self._viewbox.events.resize.connect(self.view_resize_event)
+    def link(self, camera):
+        """ Link this camera with another camera of the same type
+        
+        Linked camera's keep each-others' state in sync.
+        """
+        cam1, cam2 = self, camera
+        # Remove if already linked
+        while cam1 in cam2._linked_cameras:
+            cam2._linked_cameras.remove(cam1)
+        while cam2 in cam1._linked_cameras:
+            cam1._linked_cameras.remove(cam2)
+        # Link both ways
+        cam1._linked_cameras.append(cam2)
+        cam2._linked_cameras.append(cam1)
     
-    def disconnect(self):
-        self._viewbox.events.mouse_press.disconnect(self.view_mouse_event)
-        self._viewbox.events.mouse_release.disconnect(self.view_mouse_event)
-        self._viewbox.events.mouse_move.disconnect(self.view_mouse_event)
-        self._viewbox.events.mouse_wheel.disconnect(self.view_mouse_event)
-        self._viewbox.events.resize.disconnect(self.view_resize_event)
+    def view_changed(self):
```

Comment:
This was meant to be used when the camera is connected to a new view, not whenever the view range changes. Perhaps it needs a different name to make that clear..